includes/integrations/class-sst-deposits-for-wc.php

```diff
@@ -0,0 +1,658 @@
+			return $cart_contents;
+		}
+
+		// Don't filter unless displaying totals in cart or order review
+		if (
+			! doing_action( 'woocommerce_cart_totals_before_order_total' ) &&
+			! doing_action( 'woocommerce_review_order_before_order_total' )
+		) {
+			return $cart_contents;
+		}
+
+		// Remove filter to avoid infinite recursion
+		remove_filter(
+			'woocommerce_get_cart_contents',
+			array( $this, 'add_tax_to_totals' )
+		);
+
+		// Calculate tax due in future and update totals
+		$future_taxes = $this->calculate_future_taxes();
+
+		foreach ( $cart_contents as &$item ) {
+			if ( empty( $item['has_deposit'] ) ) {
+				continue;
+			}
+
+			$future_tax = $future_taxes[ $item['key'] ] ?? 0;
+			$unit_tax   = $future_tax / $item['quantity'];
+
+			$item['full_amount'] = $item['full_amount'] + $unit_tax;
+		}
+
+		// Restore tax totals for amount due today
+		WC()->cart->calculate_totals();
+
+		// Add removed filter back
+		add_filter(
+			'woocommerce_get_cart_contents',
+			array( $this, 'add_tax_to_totals' )
+		);
+
+		return $cart_contents;
+	}
+
+	/**
+	 * Check whether DFW created an order.
+	 *
+	 * @param WC_Order $order Order object
+	 * @return bool Did DFW create the order?
+	 */
+	public function is_dfw_order( $order ) {
+		return (
+			$order->get_meta( '_is_child_order' ) &&
+			$order->get_meta( '_original_order_id' ) &&
+			$order->get_meta( '_payment_number' )
+		);
+	}
+
+	/**
+	 * Check whether an order is a DFW duplicate order.
+	 *
+	 * @param WC_Order $order Order object
+	 * @return bool
+	 */
+	public function is_dfw_duplicate_order( $order ) {
+		return (
+			$this->is_dfw_order( $order ) &&
+			'' !== $order->get_meta( '_child_total' )
+		);
+	}
+
+	/**
+	 * Check whether DFW is creating a new order.
+	 */
+	public function is_dfw_creating_order() {
+		return (
+			! empty( $_GET['invoice_remaining_balance'] ) ||
+			! empty( $_GET['invoice_order_remaining_balance'] )
+		);
+	}
+
+	/**
+	 * Filter sst_should_calculate_order_tax to enable tax calculations
+	 * for orders created by Deposits for WooCommerce.
+	 *
+	 * NOTE: `sst_order_calculate_taxes` sets `and_taxes` to false. Returning
+	 * `and_taxes` instead of `true` prevents infinite recursion.
+	 *
+	 * @param bool     $should_calculate Whether to calculate order taxes
+	 * @param WC_Order $order            Order object
+	 * @param bool     $and_taxes        Whether to calculate order taxes
+	 *
+	 * @return bool
+	 */
+	public function should_calculate_order_tax( $should_calculate, $order, $and_taxes ) {
+		if ( $this->is_dfw_order( $order ) ) {
+			return $and_taxes;
+		}
+
+		return $should_calculate;
+	}
+
+	/**
+	 * Set a non-zero GMT offset when calculating taxes so current_time
+	 * returns a timestamp in the future.
+	 *
+	 * Without this, Stripes throws an error that the invoice due_date
+	 * is in the past if the tax calculation takes more than 1s to
+	 * to complete.
+	 */
+	public function filter_gmt_offset() {
+		return 1;
+	}
```

Comment:
I think we'll only run into an issue if the GMT offset for the site is <= 0. Maybe we should return the original value if it's greater than zero to minimize potential impacts on other plugins.